cmd/docker/docker.go

```diff
@@ -309,6 +348,22 @@ func runDocker(dockerCli *command.DockerCli) error {
 	return cmd.Execute()
 }
 
+func initializeTracing(ctx context.Context) error {
+	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}))
+
+	exporter, err := autoexport.NewSpanExporter(ctx)
+	if err != nil {
+		return fmt.Errorf("creating span exporter: %w", err)
+	}
+
+	otel.SetTracerProvider(sdktrace.NewTracerProvider(
+		sdktrace.WithSpanProcessor(sdktrace.NewBatchSpanProcessor(exporter)),
+		sdktrace.WithResource(resource.Default()),
+	))
```

Comment:
The tracer provider needs to be shut down before the process exits to ensure the recorded spans are flushed to the exporter.